FastBank.Services/BankAccountService/BankAccountService.cs

```diff
@@ -105,9 +110,113 @@ public void WithdrawAmount(Customer customer, BankAccount customerBankAccount)
+        {
+            var inquiryMsg = "Please input friend's email to transfer money. (type \"quit\" for exit):";
+            var emailTypeToInput = "Friend email:";
+            var emailFriend = _menuService.InputEmail(inquiryMsg, emailTypeToInput);
+
+            if (emailFriend == "quit")
+                return;
+
+            var friend = friends.FirstOrDefault(f => f.Value.Email == emailFriend).Value;
+
+            decimal amountToTransfer;
+            do
+            {
+                Console.WriteLine("Please write the amount for transferm to friend (type 'q' for exit):");
+                Console.Write("Transfer amount: ");
+                var inputTransferAmount = Console.ReadLine();
+                if (inputTransferAmount == "q")
+                    return;
+                if (!decimal.TryParse(inputTransferAmount, out amountToTransfer) || amountToTransfer <= 0)
+                {
+                    Console.WriteLine("Plese input correct ammount to transfer (press any key to continue...)");
+                    var keyIsEnter = Console.ReadKey();
+                    new MenuService().MoveToPreviousLine(keyIsEnter, 3);
+                }
+            }
+            while (amountToTransfer <= 0);
+
+            if (amountToTransfer > 0)
+            {
+                bool hasEnoughFunds = (customerBankAccount.Amount - amountToTransfer) < 0;
+                if (hasEnoughFunds)
+                {
+                    Console.WriteLine("You do not have enough funds to transfer (press any key to continue...)");
+                    Console.ReadKey();
+                }
+                else
+                {
+                    var friendAccount = bankAccountRepository.GetBankAccountByCustomer(new Customer(friend));
+                    if (friendAccount == null)
+                    {
+                        Console.WriteLine("You friend has not bank account. Press any key to continue...");
+                        Console.ReadKey();
+                    }
+                    else
+                    {
+                        Console.WriteLine($"Please confirm with Y transfer of {amountToTransfer} to {friend.Name} or press any other key to cancel...");
+                        var confirmKey = Console.ReadKey();
+                        if (confirmKey.KeyChar == 'Y')
+                        {
+                            friendAccount.DepositAmount(amountToTransfer);
+                            Update(friendAccount);
+                            customerBankAccount.WithdrawAmount(amountToTransfer);
+                            Update(customerBankAccount);
+                        }
+                    }
+                }
+            }
+        }
+
+        public void TransferMoneyToFriendMenu(Customer customer, BankAccount customerBankAccount)
         {
-            throw new NotImplementedException();
+            Console.Clear();
+            _menuService.Logo();
```

Comment:
_menuService.Logo() - please, let's change the name of the method. For example it must be something like _menuService.ShowLogo()

Reference: https://csharp-book.softuni.org/Content/Chapter-10-methods/method-naming/method-naming.html

Reply:
Renamed to ShowLogo()